source/CurieBleSerial.h

```diff
@@ -0,0 +1,279 @@
+            SerialConfig config_
+        )
+    {
+        _bleSerial->begin(baud_, config_);
+    };
+
+    virtual
+    bool
+        connectionReady(
+            void
+        )
+    {
+        _bleSerial->connectionReady();
+    };
+
+    virtual
+    void
+        end(
+            void
+        )
+    {
+        _bleSerial->end();
+    };
+
+    virtual
+    void
+        flush(
+            void
+        )
+    {
+        _bleSerial->flush();
+    };
+
+    virtual
+    void
+        lock(
+            void
+        )
+    {
+        _bleSerial->lock();
+    };
+
+    virtual
+    uint16_t
+        print(
+            uint8_t c_
+        )
+    {
+        return _bleSerial->print(c_);
+    };
+
+    virtual
+    uint16_t
+    print(
+          int32_t value_
+         )
+    {
+        return _bleSerial->print(value_);
+    };
+
+    virtual
+    uint16_t
+    print(
+          int32_t value_,
+          Radix base_
+         )
+    {
+        return _bleSerial->print(value_, base_);
+    };
+
+    virtual
+    uint16_t
+    print(
+          uint32_t value_
+         )
+    {
+        return _bleSerial->print(value_);
+    };
+
+    virtual
+    uint16_t
+    print(
+          uint32_t value_,
+          Radix base_
+         )
+    {
+        return _bleSerial->print(value_, base_);
+    };
+
+    virtual
+    uint16_t
+    print(
+          double value_
+         )
+    {
+        return _bleSerial->print(value_);
+    };
+
+    [Windows::Foundation::Metadata::DefaultOverloadAttribute]
+    virtual
+    uint16_t
+    print(
+          double value_,
+          int16_t decimal_place_
+         )
+    {
+        return _bleSerial->print(value_, decimal_place_);
+    };
+
+    [Windows::Foundation::Metadata::DefaultOverloadAttribute]
+    virtual
+    uint16_t
+    print(
+        const Platform::Array<uint8_t> ^buffer_
+        )
+    {
+        return _bleSerial->print(buffer_);
+    };
+
+    virtual
+    uint16_t
+        read(
+            void
+        )
+    {
+        return _bleSerial->read();
+    };
+
+    virtual
+    void
+        unlock(
+            void
+        )
+    {
+        _bleSerial->unlock();
+    };
+
+    virtual
+    uint16_t
+    write(
+        uint8_t c_
+        )
+    {
+        return _bleSerial->write(c_);
+    };
+
+    [Windows::Foundation::Metadata::DefaultOverloadAttribute]
+    virtual
+    uint16_t
+    write(
+        const Platform::Array<uint8_t> ^buffer_
+        )
+    {
+        return _bleSerial->write(buffer_);
+    };
+
+    ///<summary>
+    ///Begins an asyncronous request for all Bluetooth LE devices that are paired and may be used to attempt a device connection.
+    ///</summary>
+    static
+    Windows::Foundation::IAsyncOperation<Windows::Devices::Enumeration::DeviceInformationCollection ^> ^
+    listAvailableDevicesAsync(
+        void
+        )
+    {
+        return BleSerial::listAvailableDevicesAsync();
+    };
+
+private:
+    BleSerial^ _bleSerial;
```

Comment:
Unfortunately, it looks like this sort of object nesting prohibits creating these as const objects.